### PR TITLE
Add a Contributing Documentation section.

### PR DESCRIPTION
**Pull request by Micheletto**
r? @hwine 

Because of the work in [PR 377](https://github.com/mozilla/frost/pull/377) I believe it is no longer necessary to suggest putting the doc updates and rendered HTML in separate commits.

### CODE REVIEW

**Comment by Micheletto** [comment]
> Because of the work in [PR 377](https://github.com/mozilla/frost/pull/377) I believe it is no longer necessary to suggest putting the doc updates and rendered HTML in separate commits.

But since that hasn't landed yet: added rendered docs [here](https://github.com/mozilla/frost/pull/379/commits/0ff7293d450b047461a77f442157ebc520039fcc).

File: docs/ContributingDocumentation.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Remember that any new sections will need to be listed in the `table of contents <https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html>`_ in the `index.rst file <https://github.com/mozilla/frost/blame/master/docs/index.rst#L10-L18>`_.
+
+Build docs and read over changes
+--------------------------------
+
+build::
+
+   cd frost
+   make -C docs/ html
```
Comment (hwine) [inline docs/ContributingDocumentation.rst:29]: prior section already has us cd'd into `frost/docs`, so we don't need `cd` or the `-C docs/` (or clarify which directory we're starting from)

File: docs/ContributingDocumentation.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Read over your changes using your `favorite web browser <https://getfirefox.com>`_.
```
Comment (hwine) [inline docs/ContributingDocumentation.rst:31]: How? What do I open?

For now, I'd vote for `index.html` in the current directory.

**Review by hwine — APPROVED** [review]
just nits

File: docs/ContributingDocumentation.rst (hunk 1 above)
Comment (Micheletto) [reply docs/ContributingDocumentation.rst]: Fixed.

File: docs/ContributingDocumentation.rst (hunk 2 above)
Comment (Micheletto) [reply docs/ContributingDocumentation.rst]: Added file URL details.

**Comment by Micheletto** [comment]
@hwine re-r?

**Review by hwine — APPROVED** [review]
r+

**Pull request merged**